group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -996,12 +989,6 @@ private void append(
                             );
 
                             if (!hasRoomFor) {
-                                if (currentBatch.builder.numRecords() == 0) {
```

Comment:
got it. So this will be thrown in append. Do we want that to still be after replay?

Reply:
The check is before the replay now to ensure that the replay is align with the batches.

Reply:
Hmm I think I was unclear. The check for if a single record fits is on append which is directly after replay. Just wanted to make sure that was ok.

Reply:
We basically do the following now:
* We check if the record fits in the current batch. If it does not, we flush the current batch and create a new batch.
* We replay the record.
* We append the record to the batch.

When the batch is written, it may fail and throw the exception. In this case, we revert to the offset of the last written batch. In my steps, it is the batch flushed at step 1.